src/delay/mod.rs

```diff
@@ -0,0 +1,48 @@
+//! Delays
+
+mod syst;
+
+pub use syst::SystickDelay as Delay;
+
+mod timer;
+
+#[cfg(any(
+    feature = "stm32f401",
+    feature = "stm32f405",
+    feature = "stm32f407",
+    feature = "stm32f411",
+    feature = "stm32f412",
+    feature = "stm32f413",
+    feature = "stm32f415",
+    feature = "stm32f417",
+    feature = "stm32f423",
+    feature = "stm32f427",
+    feature = "stm32f429",
+    feature = "stm32f437",
+    feature = "stm32f439",
+    feature = "stm32f446",
+    feature = "stm32f469",
+    feature = "stm32f479"
+))]
```

Comment:
I re-used the feature gates from the `timer` module (for the TIM2 / TIM5 timer impl).